src/Symfony/Component/Mime/Part/DataPart.php

```diff
@@ -99,6 +99,11 @@ public function hasContentId(): bool
         return null !== $this->cid;
     }
 
+    public function setContentId(string $cid): void
```

Comment:
Thank you for taking over my Issue and creating PR. However, I'm not sure this is enough.
As per http://www.faqs.org/rfcs/rfc2392.html CID should be in URL format.

So I'm not sure if this method shouldn't throw an exception when cid is invalid. I think, that old SwiftMailer was complaining about it. And also this codes generates CID in something, what looks like URL

```php
private function generateContentId(): string
{
    return bin2hex(random_bytes(16)).'@symfony';
}
```

Reply:
Maybe we can check that there is at least `@` in the id and throw an exception if not? @arxeiss Would you like to submit a PR for that?

Reply:
Seems like the right exception to throw indeed.